client/src/components/graph/graph.js

```diff
@@ -295,10 +336,21 @@ class Graph extends React.Component {
       // eslint-disable-next-line react/no-did-update-set-state --- Preventing update loop via stateChanges and diff checks
       this.setState(stateChanges);
     }
+
+    // Update image URL if width or height changes
+    if (prevState.viewport.width !== viewport.width || prevState.viewport.height !== viewport.height) {
+      this.setState({
+        imageUrl: this.createImageUrl(viewport.width, viewport.height),
+      });
+    }
```

Comment:
So if the user resizes the browser window the image needs to reload from the server. I am all for a simple solution, so given the current implementation this might be a good way to good. Could you test though whether browser resizing leads to unexpected behaviour?